### PR TITLE
feat: change title of yes_no_query to part of question

### PR DESCRIPTION
**Pull request by daywiss**
Signed-off-by: David <david@umaproject.org>

Adds a utility function to parse the type of title required based on identifier. Uses this in 2 places, one in the main list of requests, and the other in the title of yes/no request details page. 

image

### CODE REVIEW

**Review by amateima — APPROVED** [review]

File: src/hooks/useIsByTransactionParams.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
-import { useMemo } from 'react';
-import { useSearchParams } from 'react-router-dom';
-
+import { useMemo } from "react";
```
Comment (Tulun) [inline src/hooks/useIsByTransactionParams.ts:1]: On a side note: this hook has a very odd name, what is "IsByTransactionParams" supposed to mean?

Might flag this for a rename

File: src/helpers/format.ts [hunk 2]
```diff
@@ -42,9 +52,45 @@ export function formatYesNoQueryString(str: string) {
...
     return formatted;
   }
 }
+
+type FormatRequestTitleParams = {
+  identifier?: string;
+  ancillaryData?: string;
+  maxTitleLength?: number;
+  defaultTitle?: string;
+};
```
Comment (Tulun) [inline src/helpers/format.ts:67]: Although not strictly required, I believe TypeScript recommends making objects like this interfaces, not types.

File: src/helpers/format.ts [hunk 3]
```diff
@@ -42,9 +52,45 @@ export function formatYesNoQueryString(str: string) {
...
+
+export function formatRequestTitle(params: FormatRequestTitleParams) {
+  const {
+    identifier,
+    ancillaryData,
+    maxTitleLength = 55,
+    defaultTitle = "Optimistic Oracle Request",
+  } = params;
```
Comment (Tulun) [inline src/helpers/format.ts:75]: I wonder if we should just pull these constants out of the function themselves, wdyt? Call out these defaults are configuable too (even if it's just above this func).

File: src/components/index/createRequestsTableCells.tsx [hunk 4]
```diff
@@ -35,26 +35,7 @@ function createRequestsTableCells(requests: oracle.types.state.RequestIndexes) {
...
-            );
-          }
-        }
-      } catch (err: any) {
-        identifier = req.identifier;
-        console.log("Error in parsing identifier", err.message);
-      }
-
+      let identifier = formatRequestTitle(req);
       const timestamp = DateTime.fromSeconds(Number(req.timestamp)).toFormat(
```
Comment (Tulun) [inline src/components/index/createRequestsTableCells.tsx:39]: Nit: can probably make this a constant, as you are not reassigning the value in the file.

File: src/helpers/format.ts (hunk 2 above)
Comment (daywiss) [reply src/helpers/format.ts:67]: oh interesting, ok

File: src/components/index/createRequestsTableCells.tsx (hunk 4 above)
Comment (daywiss) [reply src/components/index/createRequestsTableCells.tsx:39]: good call

File: src/helpers/format.ts (hunk 3 above)
Comment (Tulun) [reply src/helpers/format.ts:75]: maxTitleLength = 55. Set 55 to a constant.

Comment (daywiss) [reply src/helpers/format.ts:75]: i see ok

**Pull request merged**